gpkit/tests/t_model.py

```diff
@@ -21,6 +30,35 @@ class TestGP(unittest.TestCase):
     solver = None
     ndig = None
 
+    def test_autosweep(self):
+        from gpkit import units, ureg
+        x = Variable("x", "m**2")
+        xmin = Variable("xmin", "m")
+        m = Model(x**2, [x >= xmin**2 + units.m**2])
+
+        xmin_ = np.linspace(1, 10, 100)
+        tol = 1e-3
+        bst = autosweep_1d(m, tol, xmin, [1, 10], verbosity=0)
+        assert_logtol(bst["xmin"](xmin_), xmin_)
+        assert_logtol(bst["x"](xmin_), xmin_**2 + 1, tol)
+        assert_logtol(bst["cost"](xmin_), (xmin_**2 + 1)**2, tol)
+
+        # cvxopt: 7, mosek: 12
+        tol = 10**(-2*self.ndig+1)
+        m = Model(x**2, [x >= (xmin/3)**2, x >= (xmin/3)**0.5 * units.m**1.5])
+        bst = autosweep_1d(m, tol, xmin, [1, 10], verbosity=0)
+        self.assertAlmostEqual(mag(bst["cost"]([3]))[0], 1.0, 2*self.ndig-1)
+        x_bc = np.linspace(1, 3, 50)  # before corner
+        assert_logtol(bst["x"](x_bc), (x_bc/3)**0.5, tol)
+        assert_logtol(bst["cost"](x_bc), x_bc/3, tol)
+        x_ac = np.linspace(3, 10, 50)  # after corner
+        assert_logtol(bst["x"](x_ac), (x_ac/3)**2, tol)
+        assert_logtol(bst["cost"](x_ac), (x_ac/3)**4, tol)
+
+        if units:
+            self.assertEqual(bst["cost"](xmin_).units, ureg.m**4)
+            self.assertEqual(bst["x"](xmin_).units, ureg.m**2)
```

Comment:
great test, could probably be integrated into `t_examples` to test the example in the docs.